beacon-chain/blockchain/stategenerator/state_generator.go

```diff
@@ -7,37 +7,46 @@ import (
 	"github.com/prysmaticlabs/prysm/beacon-chain/core/state"
 	"github.com/prysmaticlabs/prysm/beacon-chain/db"
 	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
 	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/sirupsen/logrus"
 )
 
-// GenerateStateFromSlot generates state from the last finalized epoch till the specified slot.
-func GenerateStateFromSlot(ctx context.Context, db *db.BeaconDB, slot uint64) (*pb.BeaconState, error) {
-	fState, err := db.FinalizedState()
+// GenerateStateFromBlock generates state from the last finalized epoch till the specified block.
+func GenerateStateFromBlock(ctx context.Context, db *db.BeaconDB, block *pb.BeaconBlock) (*pb.BeaconState, error) {
+	fState, err := db.HistoricalStateFromSlot(block.Slot)
 	if err != nil {
 		return nil, err
 	}
 
-	if fState.Slot > slot {
+	if fState.Slot > block.Slot {
 		return nil, fmt.Errorf(
 			"requested slot %d < current slot %d in the finalized beacon state",
 			fState.Slot,
-			slot,
+			block,
 		)
 	}
 
-	pBlock, err := db.BlockBySlot(fState.Slot)
+	root, err := hashutil.HashBeaconBlock(fState.LatestBlock)
 	if err != nil {
-		return nil, fmt.Errorf("could not retrieve block: %v", err)
+		return nil, fmt.Errorf("unable to get block root %v", err)
 	}
-	root, err := hashutil.HashBeaconBlock(pBlock)
+
+	ancestorSet, err := lookUpFromFinalizedBlock(db, block, root)
 	if err != nil {
-		return nil, fmt.Errorf("could not tree hash parent block: %v", err)
+		return nil, fmt.Errorf("unable to look up block ancestors %v", err)
 	}
 
-	// run N state transitions to generate state
-	for i := fState.Slot + 1; i <= slot; i++ {
-		exists, blk, err := db.HasBlockBySlot(i)
-		if !exists {
+	logrus.Errorf("Start: Current slot %d and Finalized Epoch %d", fState.Slot, fState.FinalizedEpoch)
```

Comment:
Why are we logging Error here?

Reply:
Looks liek a debug, removing

Reply:
ah sry, about that I was for testing in runtime. There were like a 100 logs every state transition so it was difficult to know what was going on, putting in debug sounds good